torchdynamo/utils.py

```diff
@@ -245,28 +258,31 @@ def clone_input(x):
             (x.data_ptr() - buffer.data_ptr()) % 32
         ) // x.element_size()
         result = torch.as_strided(buffer, x.size(), x.stride(), cache_line_offset)
+        set_tensor_attributes(result, x, is_train)
```

Comment:
How about just doing

`result.requires_grad_(x.requires_grad)` on this line  and then remove the need of additional `is_train` argument.

Reply:
OK